app/models/accounting/qb/updater.rb

```diff
@@ -61,32 +61,49 @@ def qb_sync_for_loan_update
         end
       end
 
-      # updates single loan
+      # To update a loan, we first must extract any new qb data on its txns.
+      # Then we run the interest calculator, which depends on extracted qb data.
+      # Though interest calculation calculates balances on any loans whose interest
+      # is calculated, we call calculate balances again after the interest calculation
+      # because not all loans have their interest calculated in 'recalculate', but all loans need
+      # their balances calculated. Note: attempting to calculate balances
+      # on all loans before interest calculation has caused problems in the past.
       def update_loan(loan)
+        extract_qb_data(loan)
+        InterestCalculator.new(loan).recalculate
+        calculate_balances(loan)
+      end
+
+      private
+
+      # Extracts data for txns from the JSON in `quickbooks_data`
+      # into each Transaction's attributes and associated LineItems.
+      # Creates/deletes LineItems as needed.
+      def extract_qb_data(loan)
         if loan.transactions.present?
-          loan.transactions.standard_order.each do |txn|
-            extract_qb_data(txn)
+          loan.transactions.each do |txn|
```

Comment:
we do not need standard order for qb data extraction, since each txn's data extraction should be indep of other txns.